Projects/Src/ScriptFunc_R.pas

```
@@ -808,12 +808,40 @@ procedure ExecAndLogOutputLog(const S: String; const Error, FirstLine: Boolean;
 type
   TOnLog = procedure(const S: String; const Error, FirstLine: Boolean) of object;
 
+  TExecOutput = record
+    StdOut: PPSVariantIFC;
+    StdErr: PPSVariantIFC;
+  end;
+
 procedure ExecAndLogOutputLogCustom(const S: String; const Error, FirstLine: Boolean; const Data: NativeInt);
 begin
   var OnLog := TOnLog(PMethod(Data)^);
   OnLog(S, Error, FirstLine);
 end;
 
+{ Adapted from LoadStringsFromFile. Raises an exception on failure. }
+procedure ExecAndCaptureOutputLog(const FileName: String; Arr: PPSVariantIFC);

```

Function name seems confusing. Has Log in its name but is not related to logging? It's not even really specific to output reading but just a generic file to strings. Why doesn't the caller use LoadStringsFromFile?

---

Yes, I struggled with the name and was trying to make it fit with the others.

It was my intention to use LoadStringsFromFile but it is a nested function. If I could use it, then I would also need to raise an exception if it returned false and also delete the file, for both files.

This is quite a lot of boilerplate to add to the flow after `Stack.SetInt(PStart-6, ResultCode);` so I thought a separate procedure would be clearer. But I'll do whatever you prefer.

Perhaps `ExecCaptureFinalize` would be a better name?

---

Yes that's better. I'll see if I'll rename it some more later.